### PR TITLE
Ci/automated releasing

### PR DESCRIPTION
**Pull request by everettraven**
**Description of the change:**
- Adds a `.goreleaser.yml` file to use [GoReleaser](https://goreleaser.com/) to automate the binary+image build/push process for a release
- Adds a GitHub Action for automatically triggering a release and running GoReleaser on a tag push. It will also verify that image builds will pass on PRs as a check.
- Updates the Makefile as necessary to install and run GoReleaser and provide the needed environment variables for releases. Also needed to update the `ansible-operator` binary build target so the image build works appropriately when using the `make image-build` command to build the ansible-operator image locally.
- Removes the separate base.Dockerfile in favor of having a single Dockerfile that does everything via multi-stage builds (this makes it significantly easier to have a fully automated release process

**Motivation for the change:**
- Enable automated releases for this repo that are triggered on a tag push

**Open Question:**
- The plugin uses https://github.com/operator-framework/ansible-operator-plugins/blob/ab438547e82a4863bbc128d6d7bcebaf2e38e179/internal/version/version.go#L29 to determine the tag to use in the `FROM` directive when scaffolding the Dockerfile. Any thoughts on how we can adapt this such that we don't have to manually update this as a pre-release step? Typically this would be set as part of building the binary but the current intention is that this is used as a library by Operator-SDK so it will be whatever the string literal is. I'm having trouble with coming up with a way to resolve this without having to have a manual pre-release step of updating this.

**Checklist**

If the pull request includes user-facing changes, extra documentation is required:
- [ ] Add a new changelog fragment in `changelog/fragments` (see [`changelog/fragments/00-template.yaml`](https://github.com/operator-framework/ansible-operator-plugins/tree/master/changelog/fragments/00-template.yaml))
- [ ] Add or update relevant sections of the docs website in [`website/content/en/docs`](https://github.com/operator-framework/ansible-operator-plugins/tree/master/website/content/en/docs)

### CODE REVIEW

**Comment by everettraven** [comment]
currently e2e tests are expected to fail. We should probably get #4 merged prior to this PR so we can have passing e2e with new samples being properly generated.

/hold

File: images/ansible-operator/Dockerfile [hunk 1]
```diff
@@ -1,23 +1,67 @@
...
+COPY images/ansible-operator/Pipfile* ./
+# Instruct pip(env) not to keep a cache of installed packages,
+# to install into the global site-packages and
+# to clear the pipenv cache as well
+ENV PIP_NO_CACHE_DIR=1 \
+    PIPENV_SYSTEM=1 \
+    PIPENV_CLEAR=1
+# Ensure fresh metadata rather than cached metadata, install system and pip python deps,
+# and remove those not needed at runtime.
+# pip3~=21.1 fixes a vulnerability described in https://github.com/pypa/pip/pull/9827.
```
Comment (oceanc80) [inline images/ansible-operator/Dockerfile:22]: Probably no longer need this as we've upgraded pip

Comment (everettraven) [reply images/ansible-operator/Dockerfile]: https://github.com/operator-framework/ansible-operator-plugins/pull/5/commits/31da06c117a389e508e24a4e93b45b13a88ce304

**Review by oceanc80 — APPROVED** [review]

**Pull request merged**